R/count_teleconnections.R

```diff
@@ -149,21 +150,42 @@ count_watershed_teleconnections <- function(data_dir,
           .[["GCAM_ID"]] %>% unique() %>%
           length() -> tc_n_cropcover
 
-        # filter out where "is crop" is false and count land classes, adding 1 class to account for agriculture.
-        crop_and_landcover_types %>%
-          filter(is_crop == FALSE) %>%
-          .[["CDL_Class"]] %>% unique() %>%
-          length() -> number_landclasses_ex_ag
-
-        if (tc_n_cropcover > 0){
-          tc_n_landclasses = number_landclasses_ex_ag + 1
-        }else{
-          tc_n_landclasses = number_landclasses_ex_ag
-        }
         # TELECONNECTION - NUMBER OF FLOOD CONTROL DAMS WITHIN WATERSHED.
         flood_control_dams[watersheds_city, ] %>%
           length() -> tc_fcdam
 
+        # TELECONNECTION - RANK WATERSHED BASED ON % OF DEVELOPED/CULTIVATED AREA.
+        # Get frequencies of each type of land cover within the designated watershed.
+        mask_raster_to_polygon(cropcover_USA, watersheds_city) %>%
+          raster::freq() %>%
+          as.data.frame() -> freqdf
+        # Remove NA and all categories that are not land cover/use(water/background).
+        freqdf[!is.na(freqdf$value),] %>%
+          filter(!value %in% c(0,81,83,88,111)) -> all_land
```

Comment:
Please store `c(0, 81, 83, 88, 111)` as a vector in `constants.R`. Call it something like `cdl_non_land_ids`. Then you can simply have `filter(!value %in% cdl_non_land_ids)`. Same as idea for the vector below.